### PR TITLE
Update html documentation to include step functions

### PR DESCRIPTION
**Pull request by Elveskevtar**
Signed-off-by: Elveskevtar <kttarta@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source/commands.rst [hunk 1]
```diff
@@ -178,7 +178,36 @@ optional arguments:
...
+
+stepfunctions
+=============
+
+Creates a step function that automatically creates a cluster, runs user specified jobs, and tears the cluster down.
+
+For example:
+    cfncluster stepfunctions -b s3bucket -j path/to/jobs.config
+
+This uses the bucket name s3bucket, or creates it if it doesn't exist, to store the lambda source code and user specified jobs. To see how jobs in the jobs.config file should be specified, see // TODO :)
```
Comment (sean-smith) [inline docs/source/commands.rst]: Remove TODO and link page

Comment (sean-smith) [reply docs/source/commands.rst]: Also, you can add \`\` for `jobs.config` , see https://github.com/awslabs/cfncluster/blame/develop/docs/source/configuration.rst#L5

File: docs/source/stepfunctions.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+CfnCluster Stepfunctions
+########################
+
+Why Stepfunctions
+=================
+
+* Allows for complex workflows with CfnCluster
+* Handles cluster creation, teardown, and updates
+* Useful for conditional automated job execution
+* Has better interfacing with other AWS services
```
Comment (sean-smith) [inline docs/source/stepfunctions.rst]: Interfaces with other AWS services

File: docs/source/stepfunctions.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+.. image:: https://s3.amazonaws.com/global-cfncluster/doc-images/parallel_job.gif
+
+Using CfnCluster Step Functions
```
Comment (sean-smith) [inline docs/source/stepfunctions.rst]: Getting Started

File: docs/source/stepfunctions.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+===============================
+
+.. image:: https://s3.amazonaws.com/global-cfncluster/doc-images/command_start.gif
+
+1. Configure CfnCluster configuration file with :code:`cfncluster configure` or manually
```
Comment (sean-smith) [inline docs/source/stepfunctions.rst]: you can use \`\` for inline code

File: docs/source/stepfunctions.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    {
+        "cluster_name": "cfnclusterstepfunctions"
+    }
+
+
+.. image:: https://s3.amazonaws.com/global-cfncluster/doc-images/command_end.gif
+.. image:: https://s3.amazonaws.com/global-cfncluster/doc-images/JSON.gif
+
+Jobs Configuration Guide
```
Comment (sean-smith) [inline docs/source/stepfunctions.rst]: Jobs Config

File: docs/source/stepfunctions.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        ::
+
+            [job carrot]
+            local_path = relative/path/project
+            handler = script/path/in/project.sh
+
+    **IMPORTANT**: either :code:`s3_uri` or :code:`local_path` must be specified; not both
+
+    :code:`[job <job_name>]` optional parameters:
+        * :code:`wait_time`: How long to wait between rechecking the status of the job to see if it's completed; default = 10; range 1-240 due to scheduler limitations
```
Comment (sean-smith) [inline docs/source/stepfunctions.rst]: Period between polling on the status of the job in seconds.

File: docs/source/stepfunctions.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+        ::
+
+            [job dominos_pizza]
```
Comment (sean-smith) [inline docs/source/stepfunctions.rst]: don't use dominos... trademarks and stuff

Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: changing to [job danish]

File: docs/source/stepfunctions.rst (hunk 6 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: much better! also forgot my units 😐

File: docs/source/stepfunctions.rst (hunk 5 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: sounds good!

File: docs/source/stepfunctions.rst (hunk 3 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: are you telling my I could have saved 2 minutes while writing these docs??

File: docs/source/stepfunctions.rst (hunk 4 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: are you telling my I could have saved 2 minutes while writing these docs??

Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: that's four extra characters per inline code

Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: sigh

File: docs/source/stepfunctions.rst (hunk 3 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: I got started changing this one

File: docs/source/stepfunctions.rst (hunk 4 above)
Comment (sean-smith) [reply docs/source/stepfunctions.rst]: more than 4, you don't need `:code:`

File: docs/source/stepfunctions.rst (hunk 3 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: need a getting started section to learn how to change these docs tbh

File: docs/source/stepfunctions.rst (hunk 2 above)
Comment (Elveskevtar) [reply docs/source/stepfunctions.rst]: but now each line is not roughly the same character length :( fineeee

**Pull request merged**